lib/apiservers/engine/backends/push.go

```diff
@@ -0,0 +1,135 @@
+package backends
+
+import (
+	"fmt"
+	"io"
+	"net/url"
+	"os"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+	"golang.org/x/net/context"
+
+	"github.com/vmware/vic/lib/apiservers/engine/backends/cache"
+	vicarchive "github.com/vmware/vic/lib/archive"
+	"github.com/vmware/vic/lib/imagec"
+	"github.com/vmware/vic/pkg/trace"
+	"github.com/vmware/vic/pkg/vsphere/sys"
+
+	"github.com/docker/distribution/digest"
+	"github.com/docker/docker/api/types"
+	eventtypes "github.com/docker/docker/api/types/events"
+	"github.com/docker/docker/pkg/streamformatter"
+	"github.com/docker/docker/reference"
+)
+
+// PushImage initiates a push operation on the repository named localName.
+func (i *Image) PushImage(ctx context.Context, image, tag string, metaHeaders map[string][]string, authConfig *types.AuthConfig, outStream io.Writer) error {
+	// return fmt.Errorf("%s does not yet implement image.PushImage", ProductName())
+	defer trace.End(trace.Begin(fmt.Sprintf("%s:%s", image, tag)))
+
+	_, err := cache.ImageCache().Get(image)
```

Comment:
I think we can also remove this check since you also did it here https://github.com/vmware/vic/pull/5774/files#diff-981781345721c7111992d43025cc69d2R695